finality-aleph/src/sync/task_queue.rs

```diff
@@ -0,0 +1,108 @@
+use std::{
+    cmp::Ordering,
+    collections::BinaryHeap,
+    fmt::{Debug, Formatter},
+};
+
+use log::warn;
+use tokio::time::{sleep, Duration, Instant};
+
+use crate::sync::LOG_TARGET;
+
+#[derive(Clone, Eq, PartialEq)]
```

Comment:
Does it require `PartialEq...` only because of that `derive` clause. I don't see any reason for `T` to implement those traits. Removing it would allow to drop those constraints on `TaskQueue` as well.

Reply:
`ScheduledTask` needs to implement `Ord` and because of that it also needs to implement all `PartialOrd`, `Eq` and `PartialEq`. I guess we could manually implement `PartialEq` and `Eq` only for instant. Right now `Eq` together with `Ord` is not consistent.